test/e2e/scenarios/Cluster/cluster_validate_ref.yaml

```diff
@@ -0,0 +1,148 @@
+id: "CLUSTER_VALIDATE_REF"
+description: "In this validate we ref of cluster"
+#marks:
+#  - slow
+#  - blocked
+steps:
+  - id: "USER_CREATE"
+    description: "Create User"
+    create:
+      apiVersion: $CRD_GROUP/$CRD_VERSION
+      kind: User
+      metadata:
+        name: user$RANDOM_SUFFIX
+      spec:
+        name: user$RANDOM_SUFFIX
+        accessString: on +get
+        authenticationMode:
+          type_: password
+          passwords:
+            - key: password
+              name: $SECRET1
+    wait:
+      status:
+        conditions:
+          ACK.ResourceSynced:
+            status: "True"
+            timeout: 180
+  - id: "ACL_CREATE"
+    description: "Create new ACL"
+    create:
+      apiVersion: $CRD_GROUP/$CRD_VERSION
+      kind: ACL
+      metadata:
+        name: acl$RANDOM_SUFFIX
+      spec:
+        name: acl$RANDOM_SUFFIX
+        userNames:
+          - user$RANDOM_SUFFIX
```

Comment:
We can add references for user names too, yes?

Reply:
It could be more than one user here which means it is a slice. I talked with my mentor and he asked me to ignore this one.

Reply:
We absolutely support slices. You just provide a list of references rather than a single one. This should absolutely be added, it's really easy to add and you'll just need to modify this test slightly.

Reply:
should I just do?
`      UserNames:
        references:
          resource: UserName
          path: Spec.Name`

Reply:
How does controller know it is slice instead of a single User?

Reply:
Yes! Just like any other field reference. The generator will look at the field, see it's a list of strings and then knows to create a list of references

Reply:
Added. Hopefully test could work for it :D